### PR TITLE
[AIRFLOW-1286] Opened stdout and stderr as text streams in universal newlines mode.

### PR DESCRIPTION
**Pull request by ronfung**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- https://issues.apache.org/jira/browse/AIRFLOW-1286

### Description
- Use system default char encoding instead of hard code as "utf-8"

### Tests
- ./run_unit_tests.sh

### Commits
- code is self-explaining so no additional comment added.

### CODE REVIEW

**Comment by criccomini** [comment]
@bolkedebruin this is to fix an issue we had with Docker+Airflow+specific Python versions. Since it's touching a patch you made, I will give you a bit of time to review. LMK if you have questions.

**Comment by bolkedebruin** [comment]
Mmm we break the pattern of using utf-8 everywhere. UTF-8 should be able to contain all other character sets. What was the error your were getting as the patch / commit does not specify that?

My main question would be: what are we doing with the other decode parts and how do we test it? Maybe we should run python 3.5 on travis?

**Comment by criccomini** [comment]
@bolkedebruin the JIRA has the stack trace

File: airflow/task_runner/base_task_runner.py [hunk 1]
```diff
@@ -89,7 +90,7 @@ def __init__(self, local_task_job):
 
     def _read_task_logs(self, stream):
         while True:
-            line = stream.readline().decode('utf-8')
+            line = stream.readline().decode(sys.getdefaultencoding(), 'ignore')
```
Comment (bolkedebruin) [inline airflow/task_runner/base_task_runner.py]: It seems to me that the 'ignore' parameter is resolving the issue rather than sys.getdefaultencoding()?

**Comment by bolkedebruin** [comment]
Maybe you would like to use "universal_newlines=True" for the popen command (see security/kerberos.py" instead.

**Comment by criccomini** [comment]
LGTM merged

**Pull request closed without merging**